### PR TITLE
Fix debugging issues when some packages was installed on RN 0.61 app

### LINKED ISSUE #1123: Unable to resolve module `../Components/AccessibilityInfo/AccessibilityInfo` message
I posted this bug to react-native a few days ago: https://github.com/facebook/react-native/issues/26568

Probably it should instead be posted here, because when I run `react-native run-ios`, there are no such messages.

Also, debugging in VSCode completely stopped working after upgrading to RN 0.61 ('unverified breakpoint')...

## Software versions
- React-Native VS Code extension version: 0.11.2
- VSCode version:  1.38.1
- OS platform and version: macOS 10.14.6
- NodeJS version: 12.11.0
- React Native version: 0.61.1
- Expo SDK version (include if relevant): N/A

### LINKED ISSUE #1127: after upgrade to 0.12.0 cannot debug
## Actual Behavior
1. Attach to packager.
2. Application loaded on Iphone
3. Error on presented on Iphone:
Invalid or unexpected token

RCTFatal
__28-[RCTCxxBridge handleError:]_block_invoke
<redacted>
<redacted>
<redacted>
<redacted>
<redacted>
CFRunLoopRunSpecific
GSEventRunModal
UIApplicationMain
main
<redacted>

## Expected Behavior
1. Attach to packager.
2. Application loaded on Iphone
3. Debugger accepting breakpoints and presenting debug console.

## Software versions
- React-Native VS Code extension version: 
- VSCode version: 1.38.1
- OS platform and version: win 10
- NodeJS version: 10.11.0
- React Native version: 0.61.2
- Expo SDK version (include if relevant):

## Outputs (Include if relevant)
- Output of the Debug Console (View -> Toggle Debug Console):
```
OS: win32 x64
Adapter node: v10.11.0 x64
vscode-chrome-debug-core: 6.8.2
Launched debug configuration contains 'program' property which is deprecated and will be removed soon. Please replace it with: "cwd": "${workspaceFolder}"
Starting debugger app worker.
Established a connection with the Proxy (Packager) to the React Native application
Debugger worker loaded runtime on port 3103
```
- Output of the React-Native output channel (View -> Toggle Output -> Select React-Native in ListBox):
```
PASTE OUTPUT HERE
```
- Output of the Developer Tools console (Help -> Toggle Developer Tools -> Select Console tab):
```
abstractExtensionService.ts:393 Overwriting extension c:\Users\David\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\ms-vscode.node-debug2 with c:\Users\David\.vscode\extensions\ms-vscode.node-debug2-1.39.1.
_logMessageInConsole @ abstractExtensionService.ts:393
abstractExtensionService.ts:395 [steoates.autoimport]: Command `extension.scanNodeModules` appears multiple times in the `commands` section.
TMScopeRegistry.ts:45 Overwriting grammar scope name to file mapping for scope source.js.
Old grammar file: file:///c%3A/Users/David/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/extensions/javascript/syntaxes/JavaScript.tmLanguage.json.
New grammar file: file:///c%3A/Users/David/.vscode/extensions/mgmcdermott.vscode-language-babel-0.0.24/grammars/Babel-Language.json
register @ TMScopeRegistry.ts:45
TMScopeRegistry.ts:45 Overwriting grammar scope name to file mapping for scope source.groovy.
Old grammar file: file:///c%3A/Users/David/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/extensions/groovy/syntaxes/groovy.tmLanguage.json.
New grammar file: file:///c%3A/Users/David/.vscode/extensions/naco-siren.gradle-language-0.2.3/syntaxes/groovy.tmLanguage.json
register @ TMScopeRegistry.ts:45
TMScopeRegistry.ts:45 Overwriting grammar scope name to file mapping for scope source.graphql.
Old grammar file: file:///c%3A/Users/David/.vscode/extensions/kumar-harsh.graphql-for-vscode-1.15.1/syntaxes/graphql.json.
New grammar file: file:///c%3A/Users/David/.vscode/extensions/stpn.vscode-graphql-0.2.3/syntaxes/graphql.tmLanguage.json
register @ TMScopeRegistry.ts:45
extensionHost.ts:294 [Extension Host] debugger listening on port 46222
log.ts:173   ERR Error: ENOENT: no such file or directory, open 'c:\Users\David\AppData\Roaming\Code\logs\20191008T125621\exthost1\exthost.log': EntryNotFound (FileSystemError): Error: ENOENT: no such file or directory, open 'c:\Users\David\AppData\Roaming\Code\logs\20191008T125621\exthost1\exthost.log'
    at Object.t.createFileSystemProviderError (file:///C:/Users/David/AppData/Local/Programs/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:1477:875)
    at l.toFileSystemProviderError (file:///C:/Users/David/AppData/Local/Programs/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:1540:110)
    at l.<anonymous> (file:///C:/Users/David/AppData/Local/Programs/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:1534:798)
    at Generator.throw (<anonymous>)
    at a (file:///C:/Users/David/AppData/Local/Programs/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:35:450)
console.ts:137 [Extension Host] Finished activation
console.ts:137 [Extension Host] Spellchecker now active!
log.ts:173   ERR Error: ENOENT: no such file or directory, open 'c:\Users\David\AppData\Roaming\Code\logs\20191008T125621\exthost1\exthost.log': EntryNotFound (FileSystemError): Error: ENOENT: no such file or directory, open 'c:\Users\David\AppData\Roaming\Code\logs\20191008T125621\exthost1\exthost.log'
    at Object.t.createFileSystemProviderError (file:///C:/Users/David/AppData/Local/Programs/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:1477:875)
    at l.toFileSystemProviderError (file:///C:/Users/David/AppData/Local/Programs/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:1540:110)
    at l.<anonymous> (file:///C:/Users/David/AppData/Local/Programs/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:1534:798)
    at Generator.throw (<anonymous>)
    at a (file:///C:/Users/David/AppData/Local/Programs/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:35:450)
```

### PR DESCRIPTION
**Pull request by SounD120**
Fixes #1127 
Fixes #1123

### CODE REVIEW

File: test/debugger/sourceMap.test.ts [hunk 1]
```diff
@@ -111,9 +111,26 @@ suite("sourceMap", function() {
...
+            const expectedScriptBody = `//# sourceURL=http://localhost:8081/index.bundle?platform=android&dev=true&minify=false
+var sourceURL = '//# sourceURL=' + (hasOwnProperty.call(options, 'sourceURL') ? (options.sourceURL + '').replace(/[\\r\\n]/g, ' ') : 'lodash.templateSources[' + ++templateCounter + ']') + '\\n';
+//# sourceMappingURL=index.map\n`;
+            const sourceMap = new SourceMapUtil();
+
+            const result = sourceMap.removeSourceURL(scriptBody);
+            assert.equal(expectedScriptBody, result);
+        });
+
+        test("should not remove anything if sourceURL is not in the bottom of the bundle script body", function() {
```
Comment (SounD120) [inline test/debugger/sourceMap.test.ts]:
```suggestion
        test("should not remove anything if sourceURL is not in the end of the bundle script", function() {
```

**Comment by motiz88** [comment]
@SounD120 @RedMickey

Hi! Moti from the React Native team here. Sorry for the trouble caused by this change in Metro.

Please note that the ordering of the `sourceURL` and `sourceMappingURL` lines in Metro's output is not guaranteed, so this fix - which seems to behave differently depending on which of them appears last in the file - may stop working in the future.

I'd recommend changing the `sourceURL` regex back to the way it was, and adding the required logic elsewhere to give `sourceMappingURL` precedence over `sourceURL` if both are found (or whatever it is we actually need to do here - I haven't looked very closely at it).

As an aside, feel free to reach out to me for help on any issues involving React Native and source map generation, since it's an area I'm actively working on.

**Comment by SounD120** [comment]
Hi @motiz88 . Thanks for recommendation! We'll take a look on this. Are there any changes planned in the Metro to change the order of these lines in the near future?

**Comment by motiz88** [comment]
@SounD120 There is no plan to change this implementation detail right now, but we are always iterating on the build toolchain, and this is a type of detail we'd normally assume is safe for us to change.

**Comment by SounD120** [comment]
Thanks for details @motiz88 .

**Pull request merged**